.github/workflows/psalm.yml

```
@@ -0,0 +1,40 @@
+name: Static analysis
+
+on: [pull_request]
+
+jobs:
+  static-psalm-analysis:
+    runs-on: ubuntu-latest
+    strategy:
+      # do not stop on another job's failure
+      fail-fast: false
+      matrix:
+        ocp-version: [ 'dev-master', 'dev-stable23', 'dev-stable22' ]
+    name: Nextcloud ${{ matrix.ocp-version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+      - name: Set up php
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: 7.4
+          tools: composer:v2
+          coverage: none
+      - name: Install dependencies
+        run: composer i
+      - name: Install dependencies
+        run: composer require --dev christophwurst/nextcloud:${{ matrix.ocp-version }} --ignore-platform-reqs
```

```suggestion
        run: composer require --dev christophwurst/nextcloud:${{ matrix.ocp-version }} --ignore-platform-req=php
```

since you use composer v1 the ignorance can be limited ;-)

---

not for long anymore, so don't want to buy in on something like that.